packages/forge/src/governance/extensions/GovernorSorting.sol

```diff
@@ -0,0 +1,229 @@
+        return true;
+    }
+
+    /**
+     * @dev Accessor to sorted list of proposalIds in ascending order.
+     */
+    function sortedProposals(bool excludeDeletedProposals)
+        public
+        view
+        virtual
+        returns (uint256[] memory sortedProposalIdsReturn)
+    {
+        (uint256[] memory proposalIdList, VoteCounts[] memory proposalVoteCountsArray) =
+            excludeDeletedProposals ? allProposalTotalVotesWithoutDeleted() : allProposalTotalVotes();
+        int256[] memory netProposalVotes = new int256[](proposalIdList.length);
+        for (uint256 i = 0; i < proposalVoteCountsArray.length; i++) {
+            netProposalVotes[i] =
+                int256(proposalVoteCountsArray[i].forVotes) - int256(proposalVoteCountsArray[i].againstVotes);
+        }
+        for (uint256 i = 0; i < proposalIdList.length - 1; i++) {
+            // Only goes to length minus 1 because sorting the last item would be redundant
+            _sortItem(i, netProposalVotes, proposalIdList);
+        }
+        return proposalIdList;
+    }
+
+    /**
+     * @dev Setter for _sortedProposalIds, tiedAdjustedRankingPosition, _isTied, _lowestRanking,
+     * and _highestTiedRanking. Will only be called once and only needs to be called once because once the contest
+     * is complete these values don't change. Determines if a ranking is tied and also where the last
+     * iteration of a ranking is in the _sortedProposalIds list taking ties into account.
+     */
+    function setSortedAndTiedProposals() public virtual {
+        require(
+            state() == IGovernor.ContestState.Completed,
+            "GovernorSorting: contest must be to calculate sorted and tied proposals"
+        );
+        require(
+            setSortedAndTiedProposalsHasBeenRun == false,
+            "GovernorSorting: setSortedAndTiedProposals() has already been run and its respective values set"
+        );
+
+        _sortedProposalIds = sortedProposals(true);
+
+        int256 lastTotalVotes;
+        uint256 rankingBeingChecked = 1;
+        _highestTiedRanking = _sortedProposalIds.length + 1; // set as default so that it isn't 0 if no ties are found
+        for (uint256 i = 0; i < _sortedProposalIds.length; i++) {
+            uint256 lastSortedItemIndex = _sortedProposalIds.length - 1;
+
+            // decrement through the ascending sorted list
+            (uint256 currentForVotes, uint256 currentAgainstVotes) =
+                proposalVotes(_sortedProposalIds[lastSortedItemIndex - i]);
+            int256 currentTotalVotes = int256(currentForVotes) - int256(currentAgainstVotes);
+
+            // if on first item, set lastTotalVotes and continue
+            if (i == 0) {
+                lastTotalVotes = currentTotalVotes;
+
+                // if on last item, then the value at the current index is
+                // the last iteration of the last ranking's value
+                if (i + 1 == _sortedProposalIds.length) {
```

Comment:
So this in fact equals to `if (_sortedProposals.length == 1)`, not sure if this is intended

Reply:
Hmm, how so? My understanding is that it evaluates to `if (_sortedProposals.length == i + 1)`, or if `_sortedProposals` is 2 and `i` is 1, then it will evaluate to true, for example.

That is what happens when I test it as well as shown here.
<img width="545" alt="Screen Shot 2023-04-26 at 2 09 00 PM" src="https://user-images.githubusercontent.com/27874039/234665185-67fc55d6-cfd8-4b01-9e2c-340ff046ace4.png">

How are you evaluating this so that it evaluates to `if (_sortedProposals.length == 1)`?

Reply:
On line 186 there is a condition, that i == 0 and it's inside that block

Reply:
Ooo yes I missed that, thank you! Def can simplify this to just check if the first element is also the last.